### ISSUE TITLE
bright star masking

### ISSUE THREAD

**Issue by changhoonhahn** [pfs-cosmo/pfstarget#2]
Implement bright star masks in https://github.com/pfs-cosmo/pfstarget/blob/main/src/pfstarget/mask.py
and check that the masking is sufficient. 

For instance, [Kitanidis et al. (2020)](https://arxiv.org/pdf/1911.05714) examined the target density around bright Tycho stars: 
 ![Image](https://github.com/user-attachments/assets/f5eba1f9-7ad1-4426-9572-7487ce8797c5)

**Comment by yamada-yuka0322** [comment]
Currently, I am using the bright stellar mask from Jean Coupon(2017) https://academic.oup.com/pasj/article/70/SP1/S7/4004646
But seems to not match with the stellar mask used for the HSC s21a.

I scatter plotted the target galaxies in red with stellar masks applied as
NOT mask.i_mask_brightstar_ghost            AND
NOT mask.i_mask_brightstar_halo             AND
NOT mask.i_mask_brightstar_blooming         AND
On top of the bright stellar mask from Jean Coupon (masked region:purple, non-masked region:yellow)

We can see that there are some galaxies in the purple region, and there are also regions with no galaxies in yellow region.

![Image](https://github.com/user-attachments/assets/4092de7a-d9ad-456a-9911-d33de9ec397a)

**Comment by changhoonhahn** [comment]
Interesting. So there's definitely a discrepancy between the bright star masks. 

@jingssrs which bright star mask are we planning to use in the end?

**Comment by jingssrs** [comment]
_halo, _ghost, _blooming for sure I think. But not sure we should go as far as _any or ghost12, ghost15. 
For cosmology targets, we can always post-process by throwing away these near the bright star masks (exact region can be decided after we have the data); pre-survey issue is how to apply the bright star mask for calibration targets so that calibration will not be affected, we are still learning from pipeline team and project office. I will keep you in the loop if you are interested as well.

**Comment by yamada-yuka0322** [comment]
I think I found where the "target in stellar masked region" is coming from.
in the cuts.masking function, mask is defined as

_mask &= ~
_mask &= ~
...
But I think this actually should be

_mask |= ~
_mask |= ~

because we need to remove galaxies in star_halo or star_ghost or star_blooming

**Comment by jingssrs** [comment]
yes, it should be _mask |= ~, but we have not decided what's the best to use. Masayuki suggested _halo, _ghost. 

The bright star mask of HSC is defined here https://arxiv.org/abs/2108.13045, Section 4.2. @yamada-yuka0322  do you think you can check, by downloading the reference catalog and apply cuts as in Section 4.2.1. 

We can decide the proper bright star mask by checking a few matrix later.

**Comment by jingssrs** [comment]
I am thinking doing following should be useful when deciding the proper bright object (star, galaxies) mask to use.
1. check the ELGs target number density around the bright stars as used in the HSC PDR3 paper. 
2. S/N ratio or redshift success rate around the bright stars
pls let me know if you have time to work on this, basically we need some idea on this before S25B survey starts. let me know if you think we should find someone else to work on this.

**Comment by yamada-yuka0322** [comment]
I haven't looked into the commissioning run data, so I'm not sure about the redshift success rate, but I think I can do the first one, stacked target density around the bright stars.
